docs/DeveloperGuide.md

```diff
@@ -249,47 +249,210 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+`* *` | user | mark my tasks as done or undone | change the status of my tasks
+`* *` | user | view tasks by tags/priority/date | group my tasks based on a field of my choice
+`* *` | user | reserve dates for a task/event | block out time slots and add them upon confirmation of the time and date details
+`* *` |user| can view all tags and edit them | edit a specific tag of all tasks with that tag in one command
+`*` | user | have flexibility in entering commands | type in commands without having to remember the exact format
+`*` | user | use a keyboard shortcut to launch the program | launch the program quickly
+`*` | user | have suggestions on free slots | decide when to add a new task or shift current tasks
 
 {More to be added}
 
 ## Appendix B : Use Cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `TARS` and the **Actor** is the `user`, unless specified otherwise)
 
-#### Use case: Delete person
+#### Use case: UC01 - View help
 
 **MSS**
 
-1. User requests to list persons
-2. AddressBook shows a list of persons
-3. User requests to delete a specific person in the list
-4. AddressBook deletes the person <br>
+1. User requests to view help
+2. TARS shows a list of usage intructions<br>
+Use case ends.
+
+#### Use case: UC02- Add task
+
+**MSS**
+
+1. User requests to submit a new task
+2. TARS save the task and add the command to command history<br>
+Use case ends.
+
+**Extensions**
+
+2a. The format is invalid
+
+> 2a1. TARS shows an error message<br>
+  Use case resumes at step 1
+
+2b. The end datetime is smaller than start datetime
+
+> 2b1. TARS shows an error message<br>
+  Use case resumes at step 1
+  
+#### Use case: UC03- Delete task
+
+**MSS**
+
+1. User requests to list tasks
+2. TARS shows a list of tasks
+3. User requets to delete a specific task in the list
+4. TARS deletes the task<br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> 2a1. Use case ends
+
+3a. The given index is invalid
+
+> 3a1. TARS shows an error message<br>
+  Use case resumes at step 2
+  
+#### Use case: UC04 - Edit task
+
+**MSS**
+
+1. User requests to list tasks
+2. TARS shows a list of tasks
+3. User requests to edit a specific task in the list
+4. TARS updates the task<br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> 2a1. Use case ends
+
+3a. The given index is invalid
+
+> 3a1. TARS shows an error message<br>
+  Use case resumes at step 2
+  
+3b. The format is invalid
+
+> 3b1. TARS shows an error message<br>
+  Use case resumes at step 2
+  
+#### Use case: UC05 - Edit task by appending details
+
+**MSS**
+
+1. User requests to list tasks
+2. TARS shows a list of tasks
+3. User requests to edit a specific task in the list by appending details
+4. TARS updates the task<br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
```

Comment:
instead of just killing/stopping the process, try to give some form of user feedback ~ applies to all similar cases above and below